### ISSUE TITLE
[BUG] Similar commit hashes detected

### ISSUE THREAD

**Issue by adinhodovic** [tj-actions/changed-files#1135]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this issue exist in the latest version?

- [X] I'm using the latest release

### Describe the bug?

Getting the below error:

```
Verifying git version...
  Valid git version found: (2.40.1)
  Running on a push event...
  Error: Similar commit hashes detected: previous sha: 7336e339d879498cc3576cd6917a21b767ae4a is equivalent to the current sha: 7336e339d89879498cc3576cd6917a21b767ae4a.
  Error: Please verify that both commits are valid, and increase the fetch_depth to a number higher than 50.
  Error: Process completed with exit code 1.
```

I see similar issues but they're resolved. I've been running this action for a while year or so. It recently started failing (3 days ago last success and ran it again today). Any updates in the @v35 version that causes it?

### To Reproduce

Below config:

```
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Get Changed OpenAPI files
        id: changed-client-files
        uses: tj-actions/changed-files@v35
        with:
          files: |
            my_dir/**
```

This event runs on:

```
  release:
    types: [published]
```

And has been working, but stopped working 3 days ago. Runs fine for PRs and Pushes though.

### What OS are you seeing the problem on?

ubuntu-latest or ubuntu-20.04

### Expected behavior?

I just expected it to work with releases as well, it has been so far.

### Relevant log output

```shell
::group::changed-files-diff-sha
changed-files-diff-sha
  ##[debug]Resolving repository path: /home/runner/work/my-repo/.
  Verifying git version...
  Valid git version found: (2.40.1)
  Running on a push event...
  ##[debug]Getting HEAD SHA...
  ##[debug]Verifying the current commit SHA: 7336e339d89879498cc3576cd6917a21b767ae4a
  ##[debug]Current SHA: 7336e339d89879498cc3576cd6917a21b767ae4a
  ##[debug]Target branch 0.63.0...
  ##[debug]Current branch 0.63.0...
  ##[debug]Verifying the previous commit SHA: 7336e339d89879498cc3576cd6917a21b767ae4a
  Error: Similar commit hashes detected: previous sha: 7336e339d89879498cc3576cd6917a21b767ae4a is equivalent to the current sha: 7336e339d89879498cc3576cd6917a21b767ae4a.
  Error: Please verify that both commits are valid, and increase the fetch_depth to a number higher than 50.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jackton1** [comment]
@adinhodovic Curious is this release for an existing tag in which case I'll suggest using on tag push event.

**Comment by jackton1** [comment]
Given that release triggers aren't documented I'll recommend using tab push events instead.

**Comment by jackton1** [comment]
Fixed in #1136
